components/elements/buttons/HideShowPassword/HideShowPassword.test.ts

```diff
@@ -1,48 +1,54 @@
-// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-// @ts-expect-error
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-import { setup, $fetch } from '@nuxt/test-utils-edge'
+import { describe, expect, test } from 'vitest'
+import { mount } from '@vue/test-utils'
 
-import { describe, test as it } from 'vitest'
 import HideShowPassword from './HideShowPassword.vue'
-import { fireEvent, stubbedRender } from '@/testUtils'
 
-describe('<HideShowPassword />', async () => {
-  await setup({})
+describe('<HideShowPassword />', () => {
+  const HideShowPasswordWrapper = mount(HideShowPassword)
 
-  it('should render without crashing', () => {
-    const { unmount } = stubbedRender(HideShowPassword)
+  test('should render without crashing', () => {
+    expect(HideShowPasswordWrapper.isVisible()).toBe(true)
+  })
 
-    unmount()
+  const passwordButton = HideShowPasswordWrapper.get(
+    '[data-testid="hide-show-password-button"]',
+  )
+  const hiddenEyeIcon = HideShowPasswordWrapper.get(
+    '[data-testid="hidden-eye-icon"]',
+  )
+  const visibleEyeIcon = HideShowPasswordWrapper.get(
+    '[data-testid="visible-eye-icon"]',
+  )
+  describe('Visible eye, hidden eye, and the button', () => {
+    test('should render without crashing', () => {
+      expect(passwordButton.isVisible()).toBe(true)
+      expect(hiddenEyeIcon.isVisible()).toBe(true)
+      expect(visibleEyeIcon.isVisible()).toBe(true)
+    })
```

Comment:
Shouldn't `visibleEyeIcon` be invisible here?

Reply:
Ya so it's a bit weird to read but the `isVisible()` function doesn't check the css property `display: none;` which the `visibleEyeIcon` is being given because we are using `v-show` instead of `v-if`. `v-show` just conditionally sets `display: none;` whereas `v-if` conditionally renders the component so if it's false it's not in the document.